File: 03-frontend/angular-ecommerce/src/app/common/purchase.ts
import {Address} from "./address";
import {Customer} from "./customer";
import {Order} from "./order";
import {OrderItem} from "./order-item";

export class Purchase {
    public customer: Customer | undefined;
    public shippingAddress: Address | undefined;
    public billingAddress: Address | undefined;
    public order: Order | undefined;
    public orderItems: OrderItem[] | undefined;
  constructor() {
  }
}

File: 02-backend/spring-boot rest api/src/main/java/com/unejsi/springbootecommerce/dto/Purchase.java
package com.unejsi.springbootecommerce.dto;

import com.unejsi.springbootecommerce.entity.Address;
import com.unejsi.springbootecommerce.entity.Customer;
import com.unejsi.springbootecommerce.entity.Order;
import com.unejsi.springbootecommerce.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {

    private Customer customer;

    private Address shippingAddress;

    private Address billingAddress;

    private Order order;

    private Set<OrderItem> orderItems;
}
